### PR TITLE
Remove OrderedGraphs

### PR DESCRIPTION
**Pull request by jarrodmillman**

### CODE REVIEW

File: doc/reference/classes/index.rst [hunk 1]
```diff
@@ -41,12 +41,7 @@ Basic graph types
 
 .. note:: NetworkX uses `dicts` to store the nodes and neighbors in a graph.
    So the reporting of nodes and edges for the base graph classes will not
-   necessarily be consistent across versions and platforms.  If you need the
-   order of nodes and edges to be consistent (e.g., when writing automated
-   tests), please see :class:`~networkx.OrderedGraph`,
-   :class:`~networkx.OrderedDiGraph`, :class:`~networkx.OrderedMultiGraph`,
-   or :class:`~networkx.OrderedMultiDiGraph`, which behave like the base
-   graph classes but give a consistent order for reporting of nodes and edges.
+   necessarily be consistent across versions and platforms.
```
Comment (dschult) [inline doc/reference/classes/index.rst:44]: Should we mention that CPython is consistent across platforms?

**Review by dschult — COMMENTED** [review]
This is a potentially tricky change, so thanks for taking it on and it all looks good to me! :}  I have a small comment about possibly adding a phrase/sentence reassuring typical users that the order will be consistent across platforms within CPython version 3.7 or higher.

File: doc/reference/classes/index.rst (hunk 1 above)
Comment (jarrodmillman) [reply doc/reference/classes/index.rst]: @dschult @rossbar @MridulS  What do you think of the text in 9fd6f55? Feel free to suggest alternative text.

**Review by rossbar — APPROVED** [review]
Thanks @jarrodmillman, I don't have any concrete suggestions on how the wording could be improved, so LGTM!

**Review by MridulS — APPROVED** [review]
LGTM me too! Maybe an example could be added about it to explain what this means but I wouldn't worry too much about it :)

**Comment by dschult** [comment]
This PR also makes me wonder what examples of subclasses still exist in the repo. If we don’t point to Orderedgraph here, maybe we should point to ThinGraph?  If that is out, then we should make another example of subclassing — perhaps a subclass that uses SciPy.sparse_array as the backend with a dict-like interface.?

**Comment by rossbar** [comment]
> This PR also makes me wonder what examples of subclasses still exist in the repo. If we don’t point to Orderedgraph here, maybe we should point to ThinGraph?

The two examples that come to mind (from my limited reading of the documentation) are `ThinGraph` and `AntiGraph`, either of which would be a good subclassing example IMO (though it'd probably be worth going over them in detail).

> perhaps a subclass that uses SciPy.sparse_array as the backend with a dict-like interface.?

This would be a nice idea in general, though would also likely take a lot of thought to put together (how to handle node attributes, multiple edge attributes, etc.)

**Comment by rossbar** [comment]
> I think that the part of the doc_string that we are removing is immediately after the ThinGraph example (which doesn't appear any more in the MultiGraph/MultiDiGraph doc_string -- can we add an example for them -- or put back in the example for ThinGraph? 

AFAICT this PR doesn't remove the ThinGraph example from the Multi(Di)Graph docstrings... it was never present in the first place! I suspect this is another "base-class-docstrings-out-of-sync" issue per #5584 . +1 for adding the ThinGraph subclassing example (or similar) into the Mutlti(Di)Graph docstrings but I also agree it's a separate issue.

**Review by dschult — APPROVED** [review]
LGTM!   Thanks!

**Pull request merged**